scipy/optimize/tests/test_minimize_constrained.py

```diff
@@ -676,3 +676,109 @@ def opt(x):
         A = np.matrix(np.diag([1, 1]))
     lin_cons = LinearConstraint(A, -1, np.inf)
     minimize(opt, 2*[1], constraints = lin_cons)  # just checking that there are no errors
+
+
+class TestBoundedNelderMead(TestCase):
+
+    def test_rosen_brock0(self):
+        prob = Rosenbrock()
+        bounds = Bounds(-np.inf, np.inf)
+
+        result = minimize(prob.fun, prob.x0,
+                          method='Nelder-Mead',
+                          bounds=bounds)
+        assert_allclose(result.x, [1.0, 1.0], rtol=10 ** -3)
+
+        result_no_bounds = minimize(prob.fun, prob.x0,
+                                    method='Nelder-Mead')
+        assert_allclose(result.x, result_no_bounds.x)
+
+    def test_rosen_brock1(self):
```

Comment:
Do we need that many tests? I would try to group similar ones using `@pytest.mark.parametrize`. Just reading these, I am not sure about which test is doing what. Consider adding some comments too.

Reply:
I cleaned up the test codes using `@pytest.mark.parametrize`. Do you think it needs comments still?

Reply:
Thanks! The code is clear enough to not require comments IMO.